tests/test_img2cmap.py

```diff
@@ -117,3 +117,29 @@ def test_remove_transparency(test_image_input):
     imageconverter.remove_transparent()
     cmap = imageconverter.generate_cmap(4, "miami", 42)
     assert cmap.N == 4
+
+
+@pytest.mark.parametrize("test_image_input", images)
+def test_compute_hexcodes(test_image_input):
+    imageconverter = ImageConverter(test_image_input)
+    imageconverter.generate_cmap(4, "miami", 42)
+
+    assert imageconverter.hexcodes is not None
+    assert len(imageconverter.hexcodes) == 4
+
+
+def test_compute_optimal_hexcodes():
+    imageconverter = ImageConverter(image_urls[0])
+    imageconverter.generate_optimal_cmap(max_colors=8, random_state=42)
+    assert imageconverter.hexcodes is not None
+
+
+def test_break_kneed():
```

Comment:
this test passes because it doesn't assert, could we try something like:

```python
with pytest.raises(UserWarning):
    imageconverter = ImageConverter(image_urls[0])
    imageconverter.generate_optimal_cmap(max_colors=5, random_state=42)
````

If I'm reading the code correctly, I think that's along the lines of a decent test

Reply:
Will change. The idea of the test was to make sure no error was raised except for a UserWarning. This one looks better though :)